pol-core/pol/module/utilmod.cpp

```diff
@@ -164,11 +225,13 @@ Bscript::BObjectImp* UtilExecutorModule::mf_StrFormatTime()
     }
   }
 
-  char buffer[102];  // +2 for the \0 termination.
-  if ( strftime( buffer, sizeof buffer, format_string->data(), &time_struct ) > 0 )
-    return new String( buffer );
-  else
-    return new BError( "Format string too long." );
+  std::vector<char> buffer;
+  buffer.reserve( std::max( format_string->length() * 2, (size_t)50u ) );
+  while ( strftime( buffer.data(), buffer.capacity(), format_string->data(), &time_struct ) == 0 )
+  {
+    buffer.reserve( buffer.capacity() * 2 );
+  }
```

Comment:
Should we use `resize` rather than `reserve`?

From https://en.cppreference.com/w/cpp/container/vector/data
> If size() is 0, data() may or may not return a null pointer.

Reply:
Good catch! I guess today is UB day